petri/src/vm/openvmm/construct.rs

```diff
@@ -1134,26 +1142,41 @@ impl PetriVmConfigSetupCore<'_> {
                 )
             };
 
-            Some(ChipsetDeviceHandle {
+            Ok(Some(ChipsetDeviceHandle {
                 name: "tpm".to_string(),
-                resource: TpmDeviceHandle {
-                    ppi_store,
-                    nvram_store,
-                    refresh_tpm_seeds: false,
-                    ak_cert_type: tpm_resources::TpmAkCertTypeResource::None,
-                    register_layout,
-                    guest_secret_key: None,
-                    logger: None,
-                    is_confidential_vm: self.firmware.isolation().is_some(),
-                    // TODO: generate an actual BIOS GUID and put it here
-                    bios_guid: Guid::ZERO,
+                resource: chipset_device_worker_defs::RemoteChipsetDeviceHandle {
+                    device: TpmDeviceHandle {
+                        ppi_store,
+                        nvram_store,
+                        refresh_tpm_seeds: false,
+                        ak_cert_type: tpm_resources::TpmAkCertTypeResource::None,
+                        register_layout,
+                        guest_secret_key: None,
+                        logger: None,
+                        is_confidential_vm: self.firmware.isolation().is_some(),
+                        // TODO: generate an actual BIOS GUID and put it here
+                        bios_guid: Guid::ZERO,
+                    }
+                    .into_resource(),
+                    worker_host: self.make_device_worker("tpm").await?,
```

Comment:
Side note--I think we should add some kind of `Promise<T>` type to mesh, which would be something like:

```rust
pub enum Promise<T> {
    Ready(T),
    Pending(PendingFallibleRpc<T>),
}

impl<T> Future for Promise<T> {
       type Output = Result<T, RpcError<RemoteError>>;
}

impl<T> From<T> for Promise<T> { ... }
impl<T> From<PendingFallibleRpc<T>> for Promise<T> { ... }
```

Then we could use `Promise<T>` all over the resource types to allow the control process to parallelize a lot more of VM instantiation. The resolvers would just add a bunch of `.await.context("resolving XYZ")?` type calls and everything would naturally work out.